### PR TITLE
Update XDR definitions for Auth-next

### PR DESCRIPTION
**Pull request by 2opremio**

### CODE REVIEW

**Comment by 2opremio** [comment]
This work is blocked on https://github.com/stellar/soroban-tools/pull/385 , since we need the new token contract code to update the test.

**Comment by 2opremio** [comment]
We will also need a new Core version supporting the new token contract and that may take a while.

Another option is to merge this with the SAC test broken ...

@tsachiherman  thoughts?

**Comment by tsachiherman** [comment]
> 

Not sure why this PR is blocked on https://github.com/stellar/soroban-tools/pull/385, since compiling the code doesn't really require anything from `soroban-tools`; but the [#385](https://github.com/stellar/soroban-tools/pull/385) above is already approved, so it would get merged shortly.

for the core, we should try to use 19.7.1-1178.e352f0012.focal~soroban. If you can't make this work right away, I'd opt into disabling the unit test for now - just so it won't be a blocker. Than, we can keep working on fixing this in parallel. ( please ping me in that case, and I'll open an issue to track that work ).

**Comment by 2opremio** [comment]
> since compiling the code doesn't really require anything from soroban-tools

Not the Go code, but it affects the SAC test contract fixture which needs to be redone

> for the core, we should try to use 19.7.1-1178.e352f0012.focal~soroban.

Does that support auth next? (i.e. https://github.com/stellar/soroban-tools/pull/385 ) I doubt it since it was just merged.

> I'd opt into disabling the unit test for now - just so it won't be a blocker. Than, we can keep working on fixing this in parallel. 

Alright, I think I will need to do this. I will confirm if I do.

**Review by tsachiherman — COMMENTED** [review]
just one tiny comment.

File: xdr/scval.go [hunk 1]
```diff
@@ -129,3 +137,18 @@ func (s ScVal) Equals(o ScVal) bool {
...
+func (s ScAddress) Equals(o ScAddress) bool {
+	if s.Type != o.Type {
+		return false
+	}
+
+	switch s.Type {
+	case ScAddressTypeScAddressTypeAccount:
+		return s.AccountId.Equals(*o.AccountId)
+	case ScAddressTypeScAddressTypeContract:
+		return *(s.ContractId) == *(o.ContractId)
```
[the remark this answers is not in the document]
Comment (2opremio) [reply xdr/scval.go:150]: This is going to panic if any is nil

[the remark this answers is not in the document]
Comment (tsachiherman) [reply xdr/scval.go:150]: True, but the current alternative is segmentation fault.. which is even worse, right ?

**Comment by 2opremio** [comment]
@tamirms the `randxdr` seems to be getting stuck and timing out, any idea of what could be wrong?

**Comment by 2opremio** [comment]
Uhm  `TestReingestDB()` is also failing.

In particular, it's failing when submitting an `invokeHostFunction` transaction with: `stellar-core exception: unmarshaling did not consume whole message`

```
ERRO[2023-02-02T18:30:20.927+01:00] stellar-core exception: unmarshaling did not consume whole message  pid=95184 req=studioso.local/Qb9eOpM49k-000068 stack="[main.go:30 submitter.go:55 system.go:255 system.go:164 submit_transaction.go:158 handler.go:38 mux.go:425 server.go:2109 cors.go:190 server.go:2109 compress.go:190 server.go:2109 middleware.go:228 server.go:2109 middleware.go:111 server.go:2109 middleware.go:84 server.go:2109 xff_middleware.go:51 server.go:2109 middleware.go:48 server.go:2109 request_id.go:76 server.go:2109 middleware.go:38 server.go:2109 strip.go:25 server.go:2109 mux.go:82 server.go:2947 server.go:1991 asm_arm64.s:1172]"
```

I need to investigate ...

**Comment by 2opremio** [comment]
> This work is blocked on [stellar/soroban-tools#385](https://github.com/stellar/soroban-tools/pull/385) , since we need the new token contract code to update the SAC test and make it work.

I meant it's blocked on https://github.com/stellar/rs-soroban-env/pull/645 , sorry for the mixup

**Comment by tsachiherman** [comment]
> > This work is blocked on [stellar/soroban-tools#385](https://github.com/stellar/soroban-tools/pull/385) , since we need the new token contract code to update the SAC test and make it work.
> 
> I meant it's blocked on [stellar/rs-soroban-env#645](https://github.com/stellar/rs-soroban-env/pull/645) , sorry for the mixup

right.. I created https://github.com/stellar/go/issues/4757 to track the rest of the needed changes. I know that we can't complete the go monorepo changes right now; and that's fine.

**Comment by sreuland** [comment]
@2opremio , do you want to add  t.Skip("will fix test in #4757") on the SAC tests, so not blocked here. I'm starting on #4757. Also, have seen some transient `TestReingestDB` fails on gha builds, usually re-run of the failed step, clears.

**Comment by 2opremio** [comment]
@tsachiherman @sreuland This should be good to go once everything is green

**Review by tsachiherman — APPROVED** [review]

**Review by sreuland — APPROVED** [review]
looks good, thanks!

**Pull request merged**